app/models/miq_server/role_management.rb

```diff
@@ -50,7 +50,7 @@ def set_active_role_flags
   end
 
   def set_assigned_roles
-    self.role = VMDB::Config.new("vmdb").config[:server][:role]
+    self.role = ::Settings.server.role
```

Comment:
This one is interested...I'm curious if this will fail.  We will need to test role changes around this one.  There might be a race condition between setting the role in the UI and `::Settings` getting reloaded to actually reflect the change.